apps/src/studio/studio.js

```diff
@@ -2643,12 +2702,23 @@ Studio.allGoalsVisited = function() {
     var goal = Studio.spriteGoals_[i];
     if (!goal.finished) {
       if (protagonistSprite) {
+        var wasGoalFinished = goal.finished;
+
         goal.finished = spriteAtGoal(protagonistSprite, goal);
+
+        // If goal was just finished, then call the "when actor touches anything handler"
+        if (! wasGoalFinished && goal.finished) {
+          var allowQueueExtension = false;
+          var prefix = 'whenSpriteCollided-' + Studio.protagonistSpriteIndex + '-';
+          callHandler(prefix + 'anything', allowQueueExtension);
+        }
+
```

Comment:
@Bjvanminnen curious if you think this is a safe addition?  (it's done so that we can have the student add a 'score a point' action for each time a goal is touched.)